assembly/assembly-wsmaster-war/src/main/webapp/WEB-INF/classes/codenvy/che.properties

```diff
@@ -209,10 +209,14 @@ che.docker.ip.external=NULL
 #   - 'default':      internal address is address of docker host and ephemeral port are used
 #   - 'docker-local': internal address is address of container within docker network, and exposed ports
 #                     are used.
+#   - 'custom': The evaluation strategy may be customized through a template property.
 # The 'docker-local' strategy may be useful if a firewall prevents communication between che-server and
 # workspace containers, but will prevent communication when che-server and workspace containers are not
 # on the same Docker network.
-che.docker.server_evaluation_strategy=default
+che.docker.server_evaluation_strategy=custom
+
+che.docker.custom.external.template=<serverName>.<machineName>.<workspaceId>.<wildcardNipDomain>:<chePort>
+
```

Comment:
I fully agree with you. I had the same feeling when at first I wanted to use namespace/workspaceName but I only found workspaceId in containers.

`ws-agent.devMachine.alexander.java8.my-domain.com` is far better than `ws-agent.devMachine.workspace129898793.my-domain.com`

I just didn't wanted to make big changes to existing code to add namespace/ workspace name as container labels.
It doesn't impact the reverse proxy mechanism, only introducing new configuration properties. So we can iterate on that.